pkg/operator/controller/ingress/controller_test.go

```diff
@@ -1551,3 +1553,135 @@ func Test_IsProxyProtocolNeeded(t *testing.T) {
 		})
 	}
 }
+
+func Test_updateInfraIngressLoadBalancerIPs(t *testing.T) {
+	var (
+		awsPlatform = configv1.PlatformStatus{
+			Type: configv1.AWSPlatformType,
+		}
+		gcpPlatform = configv1.PlatformStatus{
+			Type: configv1.GCPPlatformType,
+		}
+		gcpPlatformWithDNSType = configv1.PlatformStatus{
+			Type: configv1.GCPPlatformType,
+			GCP: &configv1.GCPPlatformStatus{
+				CloudLoadBalancerConfig: &configv1.CloudLoadBalancerConfig{
+					DNSType: configv1.ClusterHostedDNSType,
+				},
+			},
+		}
+		ingresses = []corev1.LoadBalancerIngress{
+			{IP: "196.78.125.4"},
+		}
+		IngressLBIP         = configv1.IP("196.78.125.4")
+		gcpPlatformWithLBIP = configv1.PlatformStatus{
+			Type: configv1.GCPPlatformType,
+			GCP: &configv1.GCPPlatformStatus{
+				CloudLoadBalancerConfig: &configv1.CloudLoadBalancerConfig{
+					DNSType: configv1.ClusterHostedDNSType,
+					ClusterHosted: &configv1.CloudLoadBalancerIPs{
+						IngressLoadBalancerIPs: []configv1.IP{
+							IngressLBIP,
+						},
+					},
+				},
+			},
+		}
+		ingressesWithMultipleIPs = []corev1.LoadBalancerIngress{
+			{IP: "196.78.125.4"},
+			{IP: "10.10.10.4"},
+		}
+	)
+	testCases := []struct {
+		description   string
+		platform      *configv1.PlatformStatus
+		ingresses     []corev1.LoadBalancerIngress
+		expectedInfra configv1.Infrastructure
+		expectUpdated bool
+		expectError   bool
+	}{
+		{
+			description:   "nil platformStatus should cause an error",
+			platform:      nil,
+			ingresses:     []corev1.LoadBalancerIngress{},
+			expectUpdated: false,
+			expectError:   true,
+		},
+		{
+			description:   "unsupported platform should not cause an error",
+			platform:      &awsPlatform,
+			ingresses:     []corev1.LoadBalancerIngress{},
+			expectUpdated: false,
+			expectError:   false,
+		},
+		{
+			description:   "gcp platform without DNSType set",
+			platform:      &gcpPlatform,
+			ingresses:     []corev1.LoadBalancerIngress{},
+			expectUpdated: false,
+			expectError:   false,
+		},
+		{
+			description:   "gcp platform with DNSType and no LB IP",
+			platform:      &gcpPlatformWithDNSType,
+			ingresses:     []corev1.LoadBalancerIngress{},
+			expectUpdated: false,
+			expectError:   false,
+		},
+		{
+			description:   "gcp platform with DNSType and no LB IP",
```

Comment:
This test case has the same description as the previous one.  
```suggestion
			description:   "gcp platform with DNSType and no LB IP in infra config, service has 1 IP",
```